src/Workspaces/Core/Portable/Workspace/Solution/Document.cs

```diff
@@ -227,6 +257,22 @@ public bool TryGetSyntaxRoot(out SyntaxNode root)
         }
 
         /// <summary>
+        /// Only for features that absolutely must run synchronously (probably because they're
+        /// on the UI thread).  Right now, the only feature this is for is Outlining as VS will
```

Comment:
Should we not have the "right now..." comment, since I can imagine that quickly getting out of sync? Is it worth auditing our codebase for a .GetSyntaxTreeAsync().Result and moving others over? I could imagine this would also have improvements for some automatic formatting stuff which might be worth getting the benefit of.